### ISSUE TITLE
Error messages must be announced when tabbing through fields

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/vets-design-system-documentation#414]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

This is related to handling IE11 + JAWS reading errors twice in [#314](https://github.com/department-of-veterans-affairs/va.gov-team/issues/314)

## What happened

Errors are currently not announced to screen reader users while in forms mode (navigating by tab). This is because they aren't programmatically linked to their respective fields (either by including them in the <label> or <legend> or using aria-describedby. As a result, should a user return back to an error field earlier in the form OR be redirected there upon attempting to continue, they won't be able to hear the error message again unless they switch out of forms mode.

Originally documented in [#22298](https://github.com/department-of-veterans-affairs/va.gov-team/issues/22298) along with a recommended solution.

## What I expected to happen

- [ ] Expect to hear errors upon tabbing into a field with a screen reader.
- [ ] Create and/or modify unit tests for this component to include an aXe scan

### Current

```html
<div class="schemaform-field-template usa-input-error ">
  <!-- Label -->
  <label id="root_email-label" class="usa-input-error-label schemaform-label" for="root_email">
    Email address<span class="schemaform-required-span">(*Required)</span>
  </label>
  <!-- Error message -->
  <span class="usa-input-error-message" role="alert" id="root_email-error-message">
    <span class="sr-only">Error</span> 
      Please enter a valid email address
  </span>
  <!-- Field -->
  <div class="schemaform-widget-wrapper">
    <input type="email" id="root_email" name="root_email" value="pdf" >
  </div>
<div>
```

### Suggested

```html
<div class="schemaform-field-template usa-input-error ">
  <!-- Label -->
  <label id="root_email-label" class="usa-input-error-label schemaform-label" for="root_email">
    Email address<span class="schemaform-required-span">(*Required)</span>
    <!-- Error message is now within the label -->
    <span class="usa-input-error-message" role="alert" id="root_email-error-message">
    <span class="sr-only">Error</span> 
      Please enter a valid email address
    </span>
  </label>
  <!-- Field now has aria-invalid="true" -->
  <div class="schemaform-widget-wrapper">
    <input aria-invalid="true" type="email" id="root_email" name="root_email" value="pdf">
  </div>
<div>
```

## Reproducing
- Formation version:
- Device: Any
- Browser: Any

Steps to reproduce:

1. Enter https://staging.va.gov/health-care/covid-19-vaccine/stay-informed/form#content in browser
2. Start screenreading device
3. Create errors in all fields (e.g. enter abc into phone)
4. Attempt to continue
5. Tab through all fields again and confirm none of the errors are read out

https://user-images.githubusercontent.com/14154792/112867965-83bc3100-9089-11eb-866f-7c313a02b93c.mov

**Comment by 1Copenut** [comment]
Thank you @joshkimux!

**Comment by noahgelman** [comment]
This is in PR

**Comment by k80bowman** [comment]
@noahgelman just checking in. Is this something you might be able to get back to soon?

**Comment by noahgelman** [comment]
Yes, I was just looking at it yesterday.

**Comment by noahgelman** [comment]
@ddzz 
Can you review this PR?
https://github.com/department-of-veterans-affairs/vets-website/pull/19085/files

**Comment by taylorkaren** [comment]
Issue moved to [department-of-veterans-affairs/va-forms-system-core #250](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/250) via [**ZenHub**](https://www.zenhub.com/)

**Comment by taylorkaren** [comment]
@k80bowman : Can you close this issue? It's been moved to the VA Forms System Core repo to finish the work.

**Comment by k80bowman** [comment]
Closing, see note above.
